tests/unit_tests/test_wandb_run.py

```diff
@@ -454,3 +455,29 @@ def test_settings_unexpected_args_telemetry(runner, relay_server, capsys, user):
             # whose field 2 corresponds to unexpected arguments in Settings
             assert 2 in telemetry.get("11", [])
             run.finish()
+
+
+def test_artifact_wait_success(user, test_settings):
```

Comment:
I think these tests should go to `tests/unit_tests/test_wandb_artifacts_full.py`

You probably want to use the `wandb_init` fixture here instead of these two (and perhaps also the `relay_server` server one to check on things, although some extra work might be needed there to resolve artifacts-related stuff, which we have to do anyways at some point), see here for a minimal example: https://github.com/wandb/wandb/blob/master/tests/unit_tests/test_wandb.py#L324